backend/pipeline/audio_rating.py

```diff
@@ -0,0 +1,88 @@
+import pandas as pd
+import os
+import joblib
+from datasets import load_from_disk, Dataset
+
+from scipy.io.wavfile import read as read_wav
+import numpy as np
+
+from backend.pipeline import Pipeline
+from backend.data.audio_rating import AudioRatingBatchElement
+
+from backend.utils import safe_mkdir
+
+class AudioRatingPipeline(Pipeline):
+    """
+    Pipeline for rating audio samples from a folder of WAV files
+    """
+
+    def __init__(self, read_path : str, write_path : str, force_new : bool = False):
+        super().__init__()
+
+        self.read_path = read_path
+        self.write_path = write_path
+
+        # Assume read path points to directory of wav files
+        # Result will be a dataset containing rows of form
+        # ([file].wav, rating, comment, more comments...)
+
+        self.total_items = len(os.listdir(self.read_path))
```

Comment:
Should do more validation on the contents of the folder. Could be some non-wav files in there.

It would also be nice if we supported other file types like FLAC, OGG, AIFF, etc. MP3 can be weird with Python but that would also be helpful.

Reply:
A bit limited by gradio's capabilities here but added a filter so only .wav files are considered